### ISSUE TITLE
Updating the trait options in the SelectionSlider fails (ipywidgets 7.2)

### ISSUE THREAD

**Issue by KevinStrobel** [jupyter-widgets/ipywidgets#2075]
### Problem
Dear ipywidgets developers,

After creating a SelectionSlider with initial options, they can not be changed any more. The error could be circumvented if the initial options contains a list with two elements and 
```Python
psdplotslider.index = 1
```
is executed before setting the trait. Is there another way to set this trait later on?

Best regards 
Kevin 

### Code example
```Python
from IPython import display 
import ipywidgets as widgets
psdplotslider = widgets.SelectionSlider(
    options=['Selection'],
    value='Selection',
    description='region:',
    disabled=False,
    continuous_update=False,
    orientation='horizontal',
    readout=True,
    layout=widgets.Layout(width='40%'))

region_names = ['000_osci', '000_plat', '001_plat', '002_plat', '004_plat']
psdplotslider.set_trait('options', region_names)
```

### Current Output
```Python
---------------------------------------------------------------------------
TraitError                                Traceback (most recent call last)
<ipython-input-20-dd58e4399603> in <module>()
      1 region_names = ['000_osci', '000_plat', '001_plat', '002_plat', '004_plat']
----> 2 psdplotslider.options = region_names
      3 psdplotslider.set_trait('options', region_names)

/usr/local/lib/python3.6/dist-packages/traitlets/traitlets.py in __set__(self, obj, value)
    583             raise TraitError('The "%s" trait is read-only.' % self.name)
    584         else:
--> 585             self.set(obj, value)
    586 
    587     def _validate(self, obj, value):

/usr/local/lib/python3.6/dist-packages/traitlets/traitlets.py in set(self, obj, value)
    572             # we explicitly compare silent to True just in case the equality
    573             # comparison above returns something other than True/False
--> 574             obj._notify_trait(self.name, old_value, new_value)
    575 
    576     def __set__(self, obj, value):

/usr/local/lib/python3.6/dist-packages/traitlets/traitlets.py in _notify_trait(self, name, old_value, new_value)
   1137             new=new_value,
   1138             owner=self,
-> 1139             type='change',
   1140         ))
   1141 

/usr/local/lib/python3.6/dist-packages/ipywidgets/widgets/widget.py in notify_change(self, change)
    598                 # Send new state to front-end
    599                 self.send_state(key=name)
--> 600         super(Widget, self).notify_change(change)
    601 
    602     def __repr__(self):

/usr/local/lib/python3.6/dist-packages/traitlets/traitlets.py in notify_change(self, change)
   1174                 c = getattr(self, c.name)
   1175 
-> 1176             c(change)
   1177 
   1178     def _add_notifiers(self, handler, name, type):

/usr/local/lib/python3.6/dist-packages/ipywidgets/widgets/widget_selection.py in _propagate_options(self, change)
    211                 # will be recognized as a change (and pick up the new option value and label).
    212                 if self.index == 0:
--> 213                     self.index = None
    214                 self.index = 0
    215             else:

/usr/local/lib/python3.6/dist-packages/traitlets/traitlets.py in __set__(self, obj, value)
    583             raise TraitError('The "%s" trait is read-only.' % self.name)
    584         else:
--> 585             self.set(obj, value)
    586 
    587     def _validate(self, obj, value):

/usr/local/lib/python3.6/dist-packages/traitlets/traitlets.py in set(self, obj, value)
    557 
    558     def set(self, obj, value):
--> 559         new_value = self._validate(obj, value)
    560         try:
    561             old_value = obj._trait_values[self.name]

/usr/local/lib/python3.6/dist-packages/traitlets/traitlets.py in _validate(self, obj, value)
    589             return value
    590         if hasattr(self, 'validate'):
--> 591             value = self.validate(obj, value)
    592         if obj._cross_validation_lock is False:
    593             value = self._cross_validate(obj, value)

/usr/local/lib/python3.6/dist-packages/traitlets/traitlets.py in validate(self, obj, value)
   1868     def validate(self, obj, value):
   1869         if not isinstance(value, int):
-> 1870             self.error(obj, value)
   1871         return _validate_bounds(self, obj, value)
   1872 

/usr/local/lib/python3.6/dist-packages/traitlets/traitlets.py in error(self, obj, value)
    623             e = "The '%s' trait must be %s, but a value of %r was specified." \
    624                 % (self.name, self.info(), repr_type(value))
--> 625         raise TraitError(e)
    626 
    627     def get_metadata(self, key, default=None):

TraitError: The 'index' trait of a SelectionSlider instance must be an int, but a value of None <class 'NoneType'> was specified.
``` 

Python 3.6.3

**Comment by jasongrout** [comment]
It looks like that is an error that needs to be fixed, however that is only dealing with setting the slider thumb. This actually does work to change the options:

```
psdplotslider.options = ['000_osci', '000_plat', '001_plat', '002_plat', '004_plat']
```
We should fix that bug of the index being set to None, though.

**Comment by jasongrout** [comment]
@KevinStrobel - FYI, this is fixed in ipywidgets 7.3.2, released just a few minutes ago.
